include/velocypack/Slice.h

```diff
@@ -748,133 +835,41 @@ class Slice {
-        if (h == 0x13 || h == 0x14) {
-          // compact Array or Object
-          return readVariableValueLength<false>(_start + 1);
-        }
-
-        VELOCYPACK_ASSERT(h > 0x01 && h <= 0x0e && h != 0x0a);
-        if (h >= sizeof(SliceStaticData::WidthMap) / sizeof(SliceStaticData::WidthMap[0])) {
-          throw Exception(Exception::InternalError, "invalid Array/Object type");
-        }
-        return readIntegerNonEmpty<ValueLength>(_start + 1,
-                                                SliceStaticData::WidthMap[h]);
-      }
-
-      case ValueType::String: {
-        VELOCYPACK_ASSERT(h == 0xbf);
-
-        if (h < 0xbf) {
-          // we cannot get here, because the FixedTypeLengths lookup
-          // above will have kicked in already. however, the compiler
-          // claims we'll be reading across the bounds of the input
-          // here...
-          return h - 0x40;
-        }
-        
-        // long UTF-8 String
-        return static_cast<ValueLength>(
-            1 + 8 + readIntegerFixed<ValueLength, 8>(_start + 1));
-      }
-
-      case ValueType::Binary: {
-        VELOCYPACK_ASSERT(h >= 0xc0 && h <= 0xc7);
-        return static_cast<ValueLength>(
-            1 + h - 0xbf + readIntegerNonEmpty<ValueLength>(_start + 1, h - 0xbf));
-      }
-
-      case ValueType::BCD: {
-        if (h <= 0xcf) {
-          // positive BCD
-          VELOCYPACK_ASSERT(h >= 0xc8 && h < 0xcf);
-          return static_cast<ValueLength>(
-              1 + h - 0xc7 + readIntegerNonEmpty<ValueLength>(_start + 1, h - 0xc7));
-        }
-
-        // negative BCD
-        VELOCYPACK_ASSERT(h >= 0xd0 && h < 0xd7);
-        return static_cast<ValueLength>(
-            1 + h - 0xcf + readIntegerNonEmpty<ValueLength>(_start + 1, h - 0xcf));
-      }
-
-      case ValueType::Custom: {
-        VELOCYPACK_ASSERT(h >= 0xf4);
-        switch (h) {
-          case 0xf4: 
-          case 0xf5: 
-          case 0xf6: {
-            return 2 + readIntegerFixed<ValueLength, 1>(_start + 1);
-          }
-
-          case 0xf7: 
-          case 0xf8: 
-          case 0xf9:  {
-            return 3 + readIntegerFixed<ValueLength, 2>(_start + 1);
-          }
-          
-          case 0xfa: 
-          case 0xfb: 
-          case 0xfc: {
-            return 5 + readIntegerFixed<ValueLength, 4>(_start + 1);
-          }
-          
-          case 0xfd: 
-          case 0xfe: 
-          case 0xff: {
-            return 9 + readIntegerFixed<ValueLength, 8>(_start + 1);
-          }
-
-          default: {
-            // fallthrough intentional
-          }
-        }
-      }
-      default: {
-        // fallthrough intentional
-      }
-    }
+    return byteSize(start());
+  }
 
-    throw Exception(Exception::InternalError, "Invalid type for byteSize()");
+  // get the total byte size for the slice, including the head byte, excluding tags
+  ValueLength valueByteSize() const {
+    return byteSize(start());
```

Comment:
`byteSize()` and `valueByteSize()` perform the same task, so one of them (`byteSize()`) would be enough.
What's more confusing is that there is now a `value()` function that skips tags, but `valueByteSize()` doesn't.
So I suggest we either remove `valueByteSize()` or adjust its behavior so that it skips tags too.

Reply:
That looks like a typo, it should be valueStart().